src/Rules/Classes/LocalTypeAliasesRule.php

```diff
@@ -115,6 +118,17 @@ public function processNode(Node $node, Scope $scope): array
 				continue;
 			}
 
+			if ($nameScope !== null) {
+				$aliasNameResolvedType = $this->typeNodeResolver->resolve(new IdentifierTypeNode($aliasName), $nameScope->bypassTypeAliases());
+				if (!($aliasNameResolvedType instanceof ObjectType)
+					&& !($aliasNameResolvedType instanceof TemplateType) // aliases take precedence over type parameters, this is reported by other rules using TemplateTypeCheck
+					|| in_array($aliasName, ['self', 'parent'], true)
```

Comment:
1) The parentheses around `instanceof` aren't necessary, `!$foo instanceof Foo` works
2) But I'd like parentheses disambiguating `$foo && $bar || $baz`, I honestly don't know what happens :) So there should be either `($foo && $bar) || $baz` or `$foo && ($bar || $baz)`

Reply:
Ha, I guess that's some quirk of codesniffer then. When I had the disambiguating parentheses there, it reported "Useless parentheses" and since removing them helped, I didn't know it meant those parentheses around `instanceof`.